File: BookShop/BookShop/Data/Book.cs
using System;
using Data.Interfaces;

namespace Data
{
    public class Book : IBook
    {
        public int Id { get; set; }
        public double Price { get; set; }

        public string Name { get; set; }
              
        public string AuthorName { get; set; }
    }
}

File: BookShop/BookShop/ClientApp/src/app/models/Book.ts
export class Book {

  constructor(id: number, price: number, name: string, author: string, photoUrl: string)
  {
    this.id = id;
    this.price = price;
    this.name = name;
    this.authorName = author;
    this.photoUrl = photoUrl;
  }

  id: number;
  price: number;
  name: string;
  authorName: string;
  photoUrl: string;
}
